File: ImageProcessingFolder/matching_point.py
class MatchingPoint:
    def __init__(self, agent1_id, agent2_id, x1, y1, x2, y2):
        self.agent1_id = agent1_id
        self.agent2_id = agent2_id
        self.x1 = x1
        self.y1 = y1
        self.x2 = x2
        self.y2 = y2

    def get_agent1_id(self):
        return self.agent1_id

    def get_agent2_id(self):
        return self.agent2_id

    def get_x1(self):
        return self.x1

    def get_y1(self):
        return self.y1

    def get_x2(self):
        return self.x2

    def get_y2(self):
        return self.y2

File: AgentSoftware/MappingAgentSoftware/src/MatchingPoint.java
public class MatchingPoint {
    private String agent1Id;
    private Point pointAgent1;
    private String agent2Id;
    private Point pointAgent2;
    private FeatureMatchingType featureMatchingType;

    public MatchingPoint(String agent1Id, String agent2Id, Point p1, Point p2, FeatureMatchingType featureMatchingType) {
        this.agent1Id = agent1Id;
        this.pointAgent1 = p1;
        this.agent2Id = agent2Id;
        this.pointAgent2 = p2;
        this.featureMatchingType = featureMatchingType;
    }

    public String getAgent1Id() {
        return agent1Id;
    }

    public Point getPointAgent1() {
        return pointAgent1;
    }

    public String getAgent2Id() {
        return agent2Id;
    }

    public Point getPointAgent2() {
        return pointAgent2;
    }

    public FeatureMatchingType getFeatureMatchingType() {
        return featureMatchingType;
    }
}
